src/analysis/indices/utils.py

```diff
@@ -0,0 +1,77 @@
+import xarray as xr
+from typing import Dict, Any
+import numpy as np
+
+
+def rolling_cumsum(ds: xr.Dataset,
+                   rolling_window: int = 3) -> xr.Dataset:
+
+    ds_window = (
+        ds.rolling(time=rolling_window, center=True)
+        .sum()
+        .dropna(dim='time', how='all')
+    )
+
+    return ds_window
+
+
+def apply_over_period(da: xr.Dataset,
+                      func,
+                      in_variable: str,
+                      out_variable: str,
+                      time_str: str = 'month',
+                      **kwargs: Dict[Any, Any]) -> xr.Dataset:
```

Comment:
This is the type error for the kwargs
```
src/analysis/indices/utils.py:24: error: Incompatible types in assignment (expression has type "str", target has type "Dict[Any, Any]")
```

Reply:
https://stackoverflow.com/questions/37031928/type-annotations-for-args-and-kwargs

Reply:
I think this is resolved now. thanks for the link!